exercises/diffie-hellman/canonical-data.json

```diff
@@ -0,0 +1,67 @@
+{
+  "exercise": "diffie-hellman",
+  "version": "1.0.0",
+  "comments": [
+    "Optional tests to consider:",
+    "* Validation of parameters:",
+    "  Validate that `p`, `g` are valid.",
+    "  Validate that keys given as inputs are valid.",
+    "  Resources that show what happens if parameters are not validated:",
+    "  http://cryptopals.com/sets/5/challenges/34",
+    "  http://cryptopals.com/sets/5/challenges/35",
+    "* Large numbers:",
+    "  Although the calculations fundamentally do not require large numbers,",
+    "  this is a reasonable real-world use for them",
+    "  and it may be instructive to have an exercise on their use.",
+    "  Consult tracks with this exercise (such as the Go track) for possible inputs to use."
+  ],
+  "cases": [
+    {
+      "description": "private key is in range 1 .. p",
+      "property": "privateKeyIsInRange",
+      "input": {},
+      "expected": {}
+    },
+    {
+      "description": "private key is random",
+      "property": "privateKeyIsRandom",
+      "input": {},
+      "expected": {}
+    },
+    {
+      "description": "can calculate public key using private key",
+      "property": "publicKey",
+      "input": {
+        "p": 23,
+        "g": 5,
+        "privateKey": 6
+      },
+      "expected": 8
+    },
+    {
+      "description": "can calculate secret using other party's public key",
+      "property": "secret",
+      "input": {
+        "p": 23,
+        "theirPublicKey": 19,
+        "myPrivateKey": 6
+      },
+      "expected": 2
+    },
+    {
+      "description": "key exchange",
+      "property": "keyExchange",
+      "input": {
+        "p": 23,
+        "g": 5,
+        "alicePrivateKey": "privateKey(p)",
+        "bobPrivateKey": "privateKey(p)",
+        "alicePublicKey": "publicKey(p, g, alicePrivateKey)",
+        "bobPublicKey": "publicKey(p, g, bobPrivateKey)",
+        "secretA": "secret(p, bobPublicKey, alicePrivateKey)",
+        "secretB": "secret(p, alicePublicKey, bobPrivateKey)"
+      },
+      "expected": "secretA == secretB"
```

Comment:
A departure from the usual "your calculated value should equal the pre-calculated value known to the test". This is "these two values that you calculate should be equal". 

The only other instance of this I remember is https://github.com/exercism/problem-specifications/blob/master/exercises/zipper/canonical-data.json; see the sameResultFromOperations test and possibly the related discussion around https://github.com/exercism/problem-specifications/pull/934#issuecomment-335202217 and https://github.com/exercism/problem-specifications/pull/934#issuecomment-335952991

Reply:
Well, it is of course due to the randomness of the input that you cannot use fixed output values, so I think this will do fine.